src/textarea/textarea.tsx

```diff
@@ -9,25 +9,127 @@ interface Props {
+            value: '',
+            limit: 255,
+            characterLimit: 255,
+            stroke: '',
+            strokeDasharray: '',
+            outputText: '',
+            radius: 11,
+        };
+        this.onChange = this.onChange.bind(this);
+    }
+    public render() {
+        return (
+            <div className="textarea-component">
+                <textarea
+                    id={this.props.id}
+                    className={`form-control text-design ${this.state.stroke} ${
+                        this.props.isCharacterLimitEnabled ? 'character-limited' : ''
+                    } ${this.props.className || ''}`}
+                    rows={this.state.rows}
+                    onBlur={this.props.onBlur}
+                    onChange={this.onChange}
+                    placeholder={this.props.placeholder}
+                    data-qa={`textarea--element-${this.props.id}`}
+                    value={this.props.value}
+                    disabled={this.props.disabled}
+                />
+                {this.props.isCharacterLimitEnabled && (
+                    <svg>
+                        <circle id="gray" cx="50%" cy="50%" r={this.state.radius} />
+                        <circle
+                            id="colored"
+                            cx="50%"
+                            cy="50%"
+                            r={this.state.radius}
+                            className={this.state.stroke}
+                            style={{ strokeDasharray: this.state.strokeDasharray }}
+                        />
+                        <text
+                            className={this.state.stroke}
+                            x="50%"
+                            y="50%"
+                            text-anchor="middle"
+                            stroke-width="1px"
+                            dy=".3em"
+                        >
+                            {this.state.limit}
+                        </text>
+                    </svg>
+                )}
+            </div>
+        );
+    }
+    private onChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.props.onChange(event.target.value);
+        const { minRows, maxRows } = this.state;
+
+        const currentRows = this.getCurrentRows(event, minRows, maxRows);
+        this.setState({
+            value: event.target.value,
+            rows: currentRows < maxRows ? currentRows : maxRows,
+            limit: this.state.characterLimit - event.target.value.length,
+            stroke: this.state.characterLimit - event.target.value.length <= 0 ? 'error-text' : 'normal-text',
+            strokeDasharray: `${this.getPercentage(event)} 999`,
+        });
+    }
+
+    private getCurrentRows(event: React.ChangeEvent<HTMLTextAreaElement>, minRows: number, maxRows: number) {
+        if (!this.props.isCharacterLimitEnabled) {
+            return this.state.rows;
+        }
+        const textareaLineHeight = 14;
```

Comment:
Is that fix?

Reply:
yes it a constant or can be passed as props

Reply:
I mean what happens when styles (line height) change? Is that a problem?

Reply:
yeah that will be a problem it has to match the style in line-height so the row is increased based on text